addons/l10n_th/views/res_partner_view.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <data>
+        <record id="view_partner_form_inherit_l10n_th" model="ir.ui.view">
+            <field name="name">res.partner.form.inherit.l10n_th</field>
+            <field name="model">res.partner</field>
+            <field name="priority">14</field>
+            <field name="inherit_id" ref="base.view_partner_form"/>
+            <field name="arch" type="xml">
+                <field name="company_registry" position="attributes">
+                    <attribute name="invisible" add="country_code == 'TH'" separator=" or "/>
```

Comment:
One last detail, and we're good to go.
The other field was visible to regular partners as well, but it is now visible only to companies.

How about we keep the regular company registry for non-companies and use the "new one" only for companies?


```suggestion
                    <attribute name="invisible" add="(country_code == 'TH' and is_company)" separator=" or "/>
```